Fix intermittent account note failure by removing disappearing content check

mjankowski opened the pull request:
Essentially another pass at https://github.com/mastodon/mastodon/pull/34295

What I thought was happening in original was that the inline-alert element and the text within that element were being added separately and we had a timing thing where capybara would see the wrapper element but not (yet) the inner text, and fail.

Having done a variety of local browser experimenting, and putting `sleep` in various places to replicate the failure, I now think what's happening is: the DOM element is always there, but its visibility (and text content) is toggled. It has a transition/animation to disappear, with a time value set for that in the `AccountNote` jsx.

I think the failure issue here is that on some runs, after the field is submitted, the headless browser has already added/removed the "Saved" element before the rspec process can run the next line and check for that content, so it fails. I was able to replicate that by playing with various `sleep` values in between the send_keys line and the next line doing the check ... with no sleep it passes, but as I crept the value up past the transition threshold it would start to fail.

Since the original regression here was around the note update feature working AT ALL (and we preserve that coverage in the rest of this spec), and not the transition/animation/status text, I opted to just remove that portion of the spec here. It strikes me as the sort of thing better suited for a JS unit spec if we want to narrowly fiddle with the animation appearance/disappearance dynamics? Also open to other ideas on how to use capybara here, but I don't think we should go too deep on messing with sync/wait/etc internals.

mjankowski committed:
Fix intermittent account note failure by removing disappearing content check

ClearlyClaire reviewed (approved):
Looks ok to me, but the point of that check to begin with was to add some sort of client-side waiting in the test because I was afraid of the test aborting navigation before the request went through.

Basically, I added this to avoid a flaky test, but it seems like it actually added flakiness. Let's merge this and see if the test is still flaky.

mjankowski commented:
> Looks ok to me, but the point of that check to begin with was to add some sort of client-side waiting in the test because I was afraid of the test aborting navigation before the request went through.

It's sort of an aside to the change here, but I think the JS/UX feature does not wait on the request actually succeeding? I added a sleep in the controller and see the UI add/remove the "saved" text and update the actual note in the page DOM as well. There's probably another possible failure case here in that if (somehow) the spec managed to change the text, find it on the page, visit root page, visit account page -- all somehow before the controller changed the note -- the `AccountNote` assertion lower down would fail.

Not sure if "UI updates before actual change" is intended behavioür or not.

> Basically, I added this to avoid a flaky test, but it seems like it actually added flakiness. Let's merge this and see if the test is still flaky.

Sounds good.

ClearlyClaire commented:
> It's sort of an aside to the change here, but I think the JS/UX feature does not wait on the request actually succeeding? I added a sleep in the controller and see the UI add/remove the "saved" text and update the actual note in the page DOM as well. There's probably another possible failure case here in that if (somehow) the spec managed to change the text, find it on the page, visit root page, visit account page -- all somehow before the controller changed the note -- the `AccountNote` assertion lower down would fail.

Yes, that's accurate, the JS part shows “Saved” as soon as the request is issued, and does not wait for the result. So this additional code would not 100% ensure it either, it would have just made sure the request was actually fired to begin with.
